mlflow/tracing/assessment.py

```diff
@@ -0,0 +1,189 @@
+from typing import Any, Optional, Union
+
+from mlflow.entities.assessment import (
+    Assessment,
+    AssessmentError,
+    AssessmentValueType,
+    Expectation,
+    Feedback,
+    experimental,
+)
+from mlflow.entities.assessment_source import AssessmentSource
+from mlflow.exceptions import MlflowException
+from mlflow.tracking.client import MlflowClient
+
+
+@experimental
+def log_expectation(
+    trace_id: str,
+    name: str,
+    value: AssessmentValueType,
+    source: Union[str, AssessmentSource],
```

Comment:
nit: should we swap the order of the args to match `log_feedback`?

```suggestion
    source: Union[str, AssessmentSource],
    value: AssessmentValueType,
```

Reply:
Right. Tbh I think the most intuitive order is still `name` -> `value` and `source` should be optional, but it is the current backend design... I'll align them for now.